pkg/controller/persistentvolume/controller.go

```diff
@@ -1064,8 +1065,11 @@ func (ctrl *PersistentVolumeController) doDeleteVolume(volume *api.PersistentVol
 	return nil
 }
 
-// provisionClaim starts new asynchronous operation to provision a claim.
+// provisionClaim starts new asynchronous operation to provision a claim if provisioning is enabled.
 func (ctrl *PersistentVolumeController) provisionClaim(claim *api.PersistentVolumeClaim) error {
+	if !ctrl.enableDynamicProvisioning {
```

Comment:
I think this should be checked at the site that does the provisioning and this method just not invoked if it shouldn't be.